api/src/logger.ts

```diff
@@ -17,11 +18,37 @@ if (env.LOG_STYLE !== 'raw') {
 	pinoOptions.prettifier = require('pino-colada');
 }
 
-const logger = pino(pinoOptions);
+const loggerEnvConfig = getConfigFromEnv('LOGGER_', 'LOGGER_HTTP');
+
+// Expose custom log levels into formatter function
+if (loggerEnvConfig.levels) {
+	const customLogLevels: { [key: string]: string } = {};
+
+	for (const el of loggerEnvConfig.levels.split(',')) {
+		const key_val = el.split(':');
+		customLogLevels[key_val[0].trim()] = key_val[1].trim();
+	}
+
+	pinoOptions.formatters = {
+		level(label: string, number: any) {
+			return {
+				severity: customLogLevels[label] || 'info',
+				level: number,
+			};
+		},
+	};
+
+	delete loggerEnvConfig.levels;
+}
+
+const logger = pino(Object.assign(pinoOptions, loggerEnvConfig));
+
+const httpLoggerEnvConfig = getConfigFromEnv('LOGGER_HTTP', ['LOGGER_HTTP_LOGGER']);
 
```

Comment:
That's good to know! Agreed, no need for a fallback. Probably just a good idea to add the option validation then. Otherwise it may not work as expected when typo is introduced and require more detailed debugging.

Reply:
How would you validate the options?

Reply:
@JohnHillegass anything I can add/do? I'm unsure if and how I should validate options?

Reply:
Unless there's a way to dynamically pull the available options in pino itself, I don't think there's a good way for us to validate those options before passing them in. If we were to hardcode all available options and check against that, it'll be a pain to keep it up to date with the changes in the external libraries (the same applies for redis/memcache/etc/etc/etc)

Reply:
I also don't think it's needed, it's documented and linked to Pino docs so should be fine